applications/tari_indexer/src/event_manager.rs

```diff
@@ -0,0 +1,366 @@
+            })
+            .await?;
+
+        match PayloadResultStatus::try_from(response.status) {
+            Ok(PayloadResultStatus::Finalized) => {
+                let proto_decision = tari_dan_p2p::proto::consensus::Decision::try_from(response.final_decision)?;
+                let final_decision = proto_decision.try_into()?;
+                if let Decision::Commit = final_decision {
+                    Ok(Some(response.execution_result)
+                        .filter(|r| !r.is_empty())
+                        .map(|r| decode(&r))
+                        .transpose()?)
+                } else {
+                    Ok(None)
+                }
+            },
+            _ => Ok(None),
+        }
+    }
+
+    fn extract_transaction_ids_from_blocks(&self, blocks: Vec<Block>) -> HashSet<TransactionId> {
+        let mut transaction_ids = HashSet::new();
+
+        for block in blocks {
+            for command in block.commands() {
+                match command {
+                    Command::Accept(t) => {
+                        transaction_ids.insert(*t.id());
+                    },
+                    _ => {
+                        // we are only interested in confirmed transactions
+                    },
+                }
+            }
+        }
+
+        transaction_ids
+    }
+
+    async fn get_new_blocks(&self, start_block: Option<BlockId>) -> Result<Vec<Block>, anyhow::Error> {
+        let mut blocks = vec![];
+
+        let committee = self.get_all_vns().await?;
+
+        let start_block_id = match start_block {
+            Some(id) => id,
+            None => {
+                let start_block = Block::zero_block(self.network);
+                *start_block.id()
+            },
+        };
+
+        for member in committee.addresses() {
+            let resp = self.get_blocks_from_vn(member, start_block_id).await;
+
+            match resp {
+                Ok(mut vn_blocks) => {
+                    blocks.append(&mut vn_blocks);
+                },
+                Err(e) => {
+                    // We do nothing on a single VN failure, we only log it
+                    warn!(
+                        target: LOG_TARGET,
+                        "Could not get blocks from vn {}: {}",
+                        member,
+                        e
+                    );
+                },
+            };
+        }
+
+        Ok(blocks)
+    }
+
+    async fn get_all_vns(&self) -> Result<Committee<PeerAddress>, anyhow::Error> {
+        // get all the committees
+        // TODO: optimize by getting all individual CommiteeShards instead of all the VNs
+        let epoch = self.epoch_manager.current_epoch().await?;
+        let full_range = RangeInclusive::new(SubstateAddress::zero(), SubstateAddress::max());
+        let mut committee = self
+            .epoch_manager
+            .get_committee_within_shard_range(epoch, full_range)
+            .await?;
+        committee.members.shuffle(&mut OsRng);
```

Comment:
Committee has a shuffle method which means you can probably remove the `rand` dependency.
```suggestion
        committee.shuffle();
```